src/material/slider/slider.ts

```diff
@@ -289,368 +344,901 @@ export class MatSliderVisualThumb implements AfterViewInit, OnDestroy {
+    // Reason: I have found a semi-consistent way to mouse up without triggering this event.
+    '(pointerup)': '_onPointerUp($event)',
     '(blur)': '_onBlur()',
-    '(focus)': '_focus.emit()',
+    '(focus)': '_onFocus()',
   },
-  providers: [
-    {
-      provide: NG_VALUE_ACCESSOR,
-      useExisting: MatSliderThumb,
-      multi: true,
-    },
-  ],
 })
-export class MatSliderThumb implements AfterViewInit, ControlValueAccessor, OnInit, OnDestroy {
-  // ** IMPORTANT NOTE **
-  //
-  // The way `value` is implemented for MatSliderThumb doesn't follow typical Angular conventions.
-  // Normally we would define a private variable `_value` as the source of truth for the value of
-  // the slider thumb input. The source of truth for the value of the slider inputs has already
-  // been decided for us by MDC to be the value attribute on the slider thumb inputs. This is
-  // because the MDC foundation and adapter expect that the value attribute is the source of truth
-  // for the slider inputs.
-  //
-  // Also, note that the value attribute is completely disconnected from the value property.
-
-  /** The current value of this slider input. */
+export class MatSliderThumb implements OnInit, OnDestroy {
   @Input()
   get value(): number {
-    return coerceNumberProperty(this._hostElement.getAttribute('value'));
+    return coerceNumberProperty(this._hostElement.value);
   }
   set value(v: NumberInput) {
-    const value = coerceNumberProperty(v);
-
-    // If the foundation has already been initialized, we need to
-    // relay any value updates to it so that it can update the UI.
-    if (this._slider._initialized) {
-      this._slider._setValue(value, this._thumbPosition);
+    const val = coerceNumberProperty(v).toString();
+    if (this._hasSetInitialValue && !this._isActive) {
+      this._hostElement.value = val;
+      this._updateThumbUIByValue();
+      this._slider._onValueChange(this);
+      this._cdr.detectChanges();
     } else {
-      // Setup for the MDC foundation.
-      this._hostElement.setAttribute('value', `${value}`);
+      this._initialValue = val;
     }
   }
+  @Output() readonly valueChange: EventEmitter<string> = new EventEmitter<string>();
 
-  /**
-   * Emits when the raw value of the slider changes. This is here primarily
-   * to facilitate the two-way binding for the `value` input.
-   * @docs-private
-   */
-  @Output() readonly valueChange: EventEmitter<number> = new EventEmitter<number>();
+  /** The current translateX in px of the slider visual thumb. */
+  get translateX(): number {
+    if (this._slider.min >= this._slider.max) {
+      this._translateX = this._slider._inputOffset;
+      return this._translateX;
+    }
+    if (this._translateX === undefined) {
+      this._translateX = this._calcTranslateXByValue();
+    }
+    return this._translateX;
+  }
+  set translateX(v: number) {
+    this._translateX = v;
+  }
+  _translateX: number | undefined;
```

Comment:
No because it gets used in the parent MatSlider to determine if the thumbs are overlapping. Would it make sense to move `_areThumbsOverlapping` into MatSliderRangeThumb? This would allow us to make `_translateX` private?

Reply:
Don't we have a public getter for it already though?